src/main/java/Parser.java

```diff
@@ -0,0 +1,238 @@
+      }
+      tasks.get(taskNum).setAsDone();
+
+    } else if (command.equals("bye")) {
+      return;
+    } else if (command.equals("list")) {
+      taskList.printTasks();
+    } else {
+      throw new DukeException("OOPS! I'm sorry, I don't know what that means! :(");
+    }
+
+    taskList.setTaskList(tasks);
+  }
+
+  public void executeAndSave(TaskList taskList, Storage s) throws DukeException, IOException {
+
+    ArrayList<Task> tasks = taskList.getTaskList();
+
+    if (command.equals("todo")) {
+      if (taskDescription == "") {
+        throw new DukeException("OOPS! The description for todo should not be empty.");
+      }
+      ToDo task = new ToDo(taskDescription);
+      tasks.add(task);
+
+      // Printing Output
+      System.out.println("\t____________________________________________________________");
+      System.out.println("\n\tGot it! I've added this task: ");
+      System.out.println("\n\t" + task.toString());
+      System.out.println("\n\tNow you have " + tasks.size() + " tasks in the list.");
+      System.out.println("\t____________________________________________________________\n");
+
+      // Save the command in taskList.txt
+      String text = "todo " + taskDescription + "\n";
+      s.appendToFile(text);
+
+    } else if (command.equals("deadline")) {
+
+      DeadLine task = new DeadLine(taskDescription, date);
+      tasks.add(task);
+
+      System.out.println("\t____________________________________________________________");
+      System.out.println("\n\tGot it! I've added this task: ");
+      System.out.println("\n\t" + task.toString());
+      System.out.println("\n\tNow you have " + tasks.size() + " tasks in the list.");
+      System.out.println("\t____________________________________________________________\n");
+
+      // Save the command in taskList.txt
+      String text = "deadline " + taskDescription + " /by " + dateBeforeFormat + "\n";
+      s.appendToFile(text);
+
+    } else if (command.equals("event")) {
+
+      Event task = new Event(taskDescription, date);
+      tasks.add(task);
+
+      System.out.println("\t____________________________________________________________");
+      System.out.println("\n\tGot it! I've added this task: ");
+      System.out.println("\n\t" + task.toString());
+      System.out.println("\n\tNow you have " + tasks.size() + " tasks in the list.");
+      System.out.println("\t____________________________________________________________\n");
+
+      // Save the command in taskList.txt
+      String text = "event " + taskDescription + " /at " + dateBeforeFormat + "\n";
+      s.appendToFile(text);
+
+    } else if (command.equals("delete")) {
+      if (taskNum < 0 || taskNum >= tasks.size()) {
+        throw new DukeException("OOPS! Integer is out of range of list.");
+      }
+      Task removed = tasks.get(taskNum);
+      tasks.remove(taskNum);
+
+      System.out.println("\t____________________________________________________________");
```

Comment:
I think it would have been better to abstract adding the borders to a separate method so as to reduce repeated code. At least that is how i did it.